rcl/src/rcl/time.c

```diff
@@ -67,7 +67,7 @@ rcl_get_ros_time(void * data, rcl_time_point_value_t * current_time)
   if (!t->active) {
     return rcl_get_system_time(data, current_time);
   }
-  *current_time = rcl_atomic_load_uint64_t(&(t->current_time));
+  *current_time = (rcl_time_point_value_t)rcl_atomic_load_uint64_t(&(t->current_time));
```

Comment:
Hmpf, I can't say I love this.  If we look carefully at the types, `t->current_time` is of type `atomic_uint_least64_t` (https://github.com/ros2/rcl/blob/master/rcl/src/rcl/time.c#L30).  `rcl_atomic_load_uint64_t` expects a uint64 argument expects an `atomic_uint_least64_t` pointer as the input argument, so we are good so far.  Then `rcl_atomic_load_uint64_t` returns a `uint64_t` (https://github.com/ros2/rcl/blob/master/rcl/src/rcl/stdatomic_helper.h#L72), and here we get into trouble.  The next thing we are going to do is to store this into `*current_time` which is `rcl_time_point_value_t` -> `rcutils_time_point_value_t` -> `int64_t`.  So now we are storing an unsigned into a signed, potentially changing the interpretation.  I'm not sure how much it matters in this case since it is a 64-bit load, but this doesn't seem quite right.  @tfoote , any thoughts on this?